File: MarketComponents/BackEnd/src/main/java/com/backApi/demo/Dto/PedidoRequest.java
package com.backApi.demo.Dto;

import lombok.Data;
import java.util.List;

@Data
public class PedidoRequest {
    private List<ItemPedido> items;

    @Data
    public static class ItemPedido {
        private Integer productoId;
        private Integer cantidad;
    }
}

File: MarketComponents/MiTIendaApp/app/src/main/java/com/example/myapplication/data/request/PedidoRequest.kt
package com.example.myapplication.data.request

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class PedidoRequest(
    val items: List<ItemPedido>
) : Parcelable {
    @Parcelize
    data class ItemPedido(
        val productoId: Int,
        val cantidad: Int
    ) : Parcelable
}